text/3463-crates-io-policy-update.md

```diff
@@ -0,0 +1,233 @@
+the content is contained in the tarball. Please note that while we aim to keep
+the data structure somewhat stable, we can not give any stability guarantees on
+the exact database table layouts.
+
+If the index repository and the database dump are insufficient you may also use
+the crates.io API directly, though it is at the discretion of the crates.io to
+block any excessive usage. We require users of the crates.io API to limit
+themselves to a maximum of 1 request per second.
+
+We also require all API users to provide a user-agent header that allows us to
+uniquely identify your application. This allows us to more accurately monitor
+any impact your application may have on our service. Providing a user agent that
+only identifies your HTTP client library (such as `request/0.9.1`) increases the
+likelihood that we will block your traffic.
+
+It is recommended, to include contact information in your user-agent header:
+
+- Bad: `User-Agent: reqwest/0.9.1`
+- Better: `User-Agent: my_bot`
+- Best: `User-Agent: my_bot (my_bot.com/info)` or `User-Agent: my_bot (help@my_bot.com)`
+
+This allows us to contact you if we would like a change in your application's
+behavior without having to block your traffic.
+
+We reserve the right to block traffic from any client that we determine to be in
+violation of this policy or causing an impact on the integrity of our service.
+
+
+## Security
+
+Safety is one of the core principles of Rust, and to that end, we would like to
+ensure that cargo and crates.io have secure implementations. To learn more about
+disclosing security vulnerabilities for these tools, please reference the
+[Rust Security policy](https://www.rust-lang.org/policies/security) for more
+details.
+
+Note that this policy only applies to official Rust projects like crates.io and
+cargo, and not individual crates. The crates.io team and the Security Response
+working group are not responsible for the disclosure of vulnerabilities to
+specific crates, and if any issues are found, you should seek guidance from
+the individual crate owners and their specific policies instead.
+
+Thank you for taking the time to responsibly disclose any issues you find.
+
+
+## Violations and Enforcement
+
+crates.io retains full discretion to take action in response to a violation of
+these policies, including account suspension, account termination, or removal of
+content.
+
+We will however not be proactively monitoring the site for these kinds of
+violations, but instead relying on the community to draw them to our attention.
+
+While the majority of interactions between individuals in the Rust community
+falls within our policies, violations of those policies do occur at times.
+When they do, the crates.io team may need to take enforcement action to address
+the violations. In all cases, content and account deletion is permanent and there
+is no basis to reverse these moderation actions taken by the crates.io team.
```

Comment:
My understanding is that the crates.io team is part of the Rust project, so conflicts over team decisions can be handled by the moderation team:
https://github.com/rust-lang/rfcs/blob/master/text/3392-leadership-council.md#conflicts-involving-teams-or-project-members

But I'm not sure if I understand the new process correctly, or if the Rust project crates.io team is the same as the crates.io team in this RFC.

Whatever the formal process is, I would suggest giving people a way to ask for decisions to be clarified or reviewed by the team, and an external review process. In my experience, they help build confidence in the decision-making process, and they allow significant conflicts to be handled out

Reply:
you are absolutely correct with this statement.

> if the Rust project crates.io team is the same as the crates.io team in this RFC

yes, they are the same. currently not everyone in the crates.io team is involved in answering our support emails, but everyone would generally be allowed to do so if they wanted.

> I would suggest giving people a way to ask for decisions to be clarified or reviewed by the team, and an external review process.

the way to ask for clarifications/reviews is the support email address (help@crates.io). in cases where that isn't sufficient the mod team could step in.

Reply:
I agree with what you've said, but it seems inconsistent with the current wording:
> In all cases, content and account deletion is permanent and there is no basis to reverse these moderation actions taken by the crates.io team.

I suggest something like:
> Content and account deletion moderation actions taken by the crates.io team can be appealed to help@crates.io or the Rust moderation team (TODO: link or email)

(Unfortunately I'm on mobile so I can't make multi-line suggestions!)

Reply:
I would also want the appeal process expanded upon to include the following:

- How long should someone have to wait (generally) for an appeal decision to be made?
- What happens if an appeal is denied, can they try to appeal in the future or is it final?
- What happens if an appeal is granted? Is everything restored back, including stats?

A lot of people will gauge a package's popularity by "downloads over time" (similar to GitHub stars). Imagine having a top 10 package and then survive an appeal, get restored, and have 0 downloads over time. Is that fair?